examples/text_to_image/train_text_to_image.py

```diff
@@ -697,6 +721,51 @@ def collate_fn(examples):
             if global_step >= args.max_train_steps:
                 break
 
+        if accelerator.is_main_process:
+            if args.validation_prompt is not None and epoch % args.validation_epochs == 0:
+                logger.info(
+                    f"Running validation... \n Generating {args.num_validation_images} images with prompt:"
+                    f" {args.validation_prompt}."
+                )
+                # create pipeline
+                safety_checker = StableDiffusionSafetyChecker.from_pretrained(
+                    args.pretrained_model_name_or_path, subfolder="safety_checker", revision=args.non_ema_revision
+                )
+                # safety_checker.to(accelerator.device, dtype=weight_dtype)
+                pipeline = StableDiffusionPipeline.from_pretrained(
+                    args.pretrained_model_name_or_path,
+                    unet=accelerator.unwrap_model(unet),
```

Comment:
When doing `ema`, we should use `ema` weights for inference.

Reply:
to do that, we'll need to 

- temporarily store the `non-ema` weights
- copy the `ema` weights to `unet`
- restore the `non-ema` weight back in the `unet`.

For that we'll need to add the `store`, `restore` method in `EMAModel` as defined in https://github.com/fadel/pytorch_ema/blob/master/torch_ema/ema.py#L139

Happy to take care of this if you want :)